File: shared/core/entities/Device.ts
import { ModelName, DBObject, Schema, Types } from '../@pyro/db';
import { IDeviceRawObject, IDeviceCreateObject } from '../interfaces/IDevice';
import IPlatform from '../interfaces/IPlatform';
import ILanguage from '../interfaces/ILanguage';
import { Entity, Column } from 'typeorm';

/**
 * Represent Device (e.g. Table or Browser, used by someone to access one of our apps)
 *
 * @class Device
 * @extends {DBObject<IDeviceRawObject, IDeviceCreateObject>}
 * @implements {IDeviceRawObject}
 */
@ModelName('Device')
@Entity({ name: 'devices' })
class Device extends DBObject<IDeviceRawObject, IDeviceCreateObject>
	implements IDeviceRawObject {
	/**
	 * ChannelID used for push notifications
	 *
	 * @type {(string | null)}
	 * @memberof Device
	 */
	@Schema({ type: String, required: false })
	@Column()
	channelId: string | null;

	/**
	 * Platform of the device (broser, ios, android, etc)
	 *
	 * @type {IPlatform}
	 * @memberof Device
	 */
	@Types.String()
	@Column()
	type: IPlatform;

	/**
	 * Language setting on the device
	 *
	 * @type {ILanguage}
	 * @memberof Device
	 */
	@Types.String('en-US')
	@Column()
	language: ILanguage;

	/**
	 * Unique device Id
	 *
	 * @type {string}
	 * @memberof Device
	 */
	@Types.String()
	@Column()
	uuid: string;

	@Types.Boolean(false)
	@Column()
	isDeleted: boolean;
}

export default Device;

File: packages/shop-mobile-flutter/lib/models/device.dart
class Device {
  final String? typename;
  final String id;
  final String? channelId;
  final String type;
  final String uuid;
  final String? language;

  const Device({
    this.typename,
    required this.id,
    this.channelId,
    required this.type,
    required this.uuid,
    this.language,
  });

  factory Device.fromJson(Map<String, dynamic> json) {
    return Device(
      typename: json['__typename'],
      id: json['_id'],
      channelId: json['channelId'],
      type: json['type'],
      uuid: json['uuid'],
      language: json['language'],
    );
  }
}
